src/sys.rs

```diff
@@ -0,0 +1,425 @@
+
+pub use self::inner::*;
+
+#[cfg(unix)]
+mod inner {
+    use libc::{self, c_int, c_long, c_char, time_t};
+    use ::Tm;
+
+    #[cfg(any(target_os = "macos", target_os = "ios"))]
+    pub use self::mac::*;
+    #[cfg(all(not(target_os = "macos"), not(target_os = "ios")))]
+    pub use self::unix::*;
+
+    /// ctime's `tm`
+    #[repr(C)]
+    struct tm {
+        tm_sec: c_int,
+        tm_min: c_int,
+        tm_hour: c_int,
+        tm_mday: c_int,
+        tm_mon: c_int,
+        tm_year: c_int,
+        tm_wday: c_int,
+        tm_yday: c_int,
+        tm_isdst: c_int,
+        tm_gmtoff: c_long,
+        tm_zone: *const c_char,
+    }
+
+    impl Default for tm {
```

Comment:
Could you consolidate the logic into this function? It's a little unfortunate having it spread out over so many different areas.